application/helpers/frontend_helper.php

```diff
@@ -1897,20 +1897,30 @@ function doAssessment($surveyid, $returndataonly=false)
                 }
 
                 // If this is a question (and not a survey field, like ID), save asessment value
-                if (!empty($field['title']))
+                if ($field['qid'] > 0)
                 {
                     /**
                      * Allow Plugin to update assessment value
                      */
                     // Prepare Event Info
                     $event = new PluginEvent('afterSurveyQuestionAssessment');
                     $event->set('surveyId', $surveyid);
-                    $event->set('question', $field);
+                    $event->set('lang', $_SESSION['survey_'.$surveyid]['s_lang']);
+                    $event->set('gid', $field);
+                    $event->set('qid', $field['gid']);
+
+                    if (array_key_exists('sqid', $field))
+                    {
+                        $event->set('sqid', $field['sqid']);
+                    }
+
```

Comment:
It's not in field ? 

Arry/text + Array/number have 2 sub question : one for X and one for Y. Here you get only the Y subquestion. The X are in aid (maybe ????) 

field are really a broken thing : i really think we must not use it .....